s3/src/main/scala/akka/stream/alpakka/s3/javadsl/S3.scala

```diff
@@ -483,6 +483,27 @@ object S3 {
       .listBucket(bucket, prefix.asScala, s3Headers)
       .asJava
 
+  /**
+   * Will return a source of object metadata and common prefixes for a given bucket with optional prefix and delimiter using version 2 of the List Bucket API.
+   * This will automatically page through all keys with the given parameters.
+   *
+   * The `alpakka.s3.list-bucket-api-version` can be set to 1 to use the older API version 1
+   *
+   * @see https://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html  (version 1 API)
+   * @see https://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketGET.html (version 1 API)
+   * @see https://docs.aws.amazon.com/AmazonS3/latest/dev/ListingKeysHierarchy.html (prefix and delimiter documentation)
+   * @param bucket    Which bucket that you list object metadata for
+   * @param prefix    Prefix of the keys you want to list under passed bucket
+   * @param delimiter Delimiter to use for listing only one level of hierarchy
+   * @param s3Headers any headers you want to add
+   * @return [[akka.stream.scaladsl.Source Source]] of [[ListBucketResultBase]]
+   */
+  def listObjects(bucket: String,
```

Comment:
It seems that this should be another `listBucket` overload.  Is there a reason you gave it a different name?

Reply:
Or better yet, `listBucketWithPrefixes` to accommodate the different return type.

Reply:
listBucketWithPrefixes could work. Then we could make the delimiter parameter even non-optional. 

I've taken the listObjects name from the actual S3 API Naming as I was shortly confused myself which alpakka S3 function is linked to which API call. But I guess this only makes sense if in the long run the listBucket function gets deprecated and completely replaced by listObjects*. Until then it would probably just confuse more. So yea lets make it listBucketWithPrefixes

Reply:
Ahh it seems this "list contents/prefixes" endpoint used to be called "GET Bucket (List Objects)" in the documentation but has been renamed to "ListObjectsV2". Now I get where the original name is coming from